open_catalyst_dataset/adsorptions.py

```diff
@@ -400,6 +400,31 @@ def tag_surface_atoms(bulk_atoms, surface_atoms):
 
     tags = [1 if i in surface_indices else 0
             for i, _ in enumerate(surface_atoms)]
+
+    # In addition to voronoi detection, some elements (esp S/F/Se)
+    # still get flagged as bulk while on the surface. Adding an additional 
+    # check for any atoms that are within two angstroms of the top most atom
+    # in the direction of the second unit cell vector seems to help
+    height_tags = []
+    scaled_positions = surface_atoms.get_scaled_positions()
+    unit_cell_height = np.linalg.norm(surface_atoms.cell[2])    
+
+    # Get the max height in the unit cell
+    max_height = max(position[2] for position, atom in zip(scaled_positions, surface_atoms)
+                      if atom.tag == 0)
```

Comment:
Should it be `if atom.tog <= 1`? As-is, this `max_height` is looking only at bulk-like atoms and ignoring surface atoms.